tests/api/mam_test.c

```diff
@@ -163,25 +165,21 @@ void test_write_until_next_channel(void) {
       TEST_ASSERT_TRUE(strstr(json_result, substr));
     }
 
-    // Deserialize the response, and compare whether the payload list contains correct number of message under a certain
-    // channel ID.
-    int len = 0;
-    cJSON *json_obj = cJSON_Parse(json_result), *current_obj = NULL;
-    cJSON* json_item = cJSON_GetObjectItemCaseSensitive(json_obj, "payload");
-    TEST_ASSERT_TRUE(cJSON_IsArray(json_item));
-    cJSON_ArrayForEach(current_obj, json_item) {
-      if (current_obj->valuestring != NULL) {
-        len++;
-      }
+    recv_mam_res_deserialize(json_result, res);
+
+    if (chid1[0]) {
```

Comment:
`chid1` is not assigned with any value after the initialization?

Reply:
At line 154, I have initialized it

Reply:
But what is this for?
I mean you do not assign any value in it except the initialization.
I do not know the purpose of the `if` condition.

Reply:
There would be no `chid1`, before we announce the next channel (chid1).
In case the error happened here, we need a `if`.

Reply:
Since the code of assigning value to `child1` is added, we can resolve the conversation.